### PR TITLE
docs: Add migration guide from v4 to v5 to protocol-kit docs

### PR DESCRIPTION
**Pull request by DaniSomoza**
Add migration guide from v4 to v5 to protocol-kit docs

### CODE REVIEW

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+const protocolKit = await safeFactory.deploySafe({
+  safeAccountConfig,
+  saltNonce // Optional
+})
+
+// Confirm the Safe is deployed and fetch properties
+console.log('Is Safe deployed:', await protocolKit.isSafeDeployed());
+console.log('Safe Address:', await protocolKit.getAddress());
+console.log('Safe Owners:', await protocolKit.getOwners());
+console.log('Safe Threshold:', await protocolKit.getThreshold());
```
Comment (dasanra) [inline pages/sdk/protocol-kit/guides/migrate-to-v5.md:35]:
```suggestion
console.log('Is Safe deployed:', await protocolKit.isSafeDeployed())
console.log('Safe Address:', await protocolKit.getAddress())
console.log('Safe Owners:', await protocolKit.getOwners())
console.log('Safe Threshold:', await protocolKit.getThreshold())
```

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+const txReceipt = await client.waitForTransactionReceipt({ hash: txHash })
+
+// Reconnect to the newly deployed Safe using the protocol-kit
+protocolKit = await protocolKit.connect({ safeAddress })
+
+// Confirm the Safe is deployed and fetch properties
+console.log('Is Safe deployed:', await protocolKit.isSafeDeployed());
+console.log('Safe Address:', await protocolKit.getAddress());
+console.log('Safe Owners:', await protocolKit.getOwners());
+console.log('Safe Threshold:', await protocolKit.getThreshold());
```
Comment (dasanra) [inline pages/sdk/protocol-kit/guides/migrate-to-v5.md:82]:
```suggestion
console.log('Is Safe deployed:', await protocolKit.isSafeDeployed())
console.log('Safe Address:', await protocolKit.getAddress())
console.log('Safe Owners:', await protocolKit.getOwners())
console.log('Safe Threshold:', await protocolKit.getThreshold())
```

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  predictSafe
+})
+
+// you can predict the address of your Safe if the Safe version is `v1.3.0` or above
+const safeAddress = await protocolKit.getAddress()
+
+const deploymentTransaction = await protocolKit.createSafeDeploymentTransaction()
+
+// Execute this transaction using your Ethereum client
+const txHash = await client.sendTransaction({
```
Comment (dasanra) [inline pages/sdk/protocol-kit/guides/migrate-to-v5.md:67]: You can get a client from protocolKit

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+// old v4 code
+import { SafeFactory } from '@safe-global/protocol-kit'
+
+const safeFactory = await SafeFactory.init({
+   provider,
+   signer,
+   safeVersion // Optional
+})
+
+const safeAccountConfig = {
```
Comment (germartinez) [inline pages/sdk/protocol-kit/guides/migrate-to-v5.md:21]: Could you specify the type and add the right import?

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+// Confirm the Safe is deployed and fetch properties
+console.log('Is Safe deployed:', await protocolKit.isSafeDeployed());
+console.log('Safe Address:', await protocolKit.getAddress());
+console.log('Safe Owners:', await protocolKit.getOwners());
+console.log('Safe Threshold:', await protocolKit.getThreshold());
+```
+
+### Predict the Safe Address
+
+Before the Safe deployment, and if you are using Safe version `v1.3.0` or above, you can predict the address of your Safe account by replacing the `SafeFactory.predictSafeAddress` method with the `Safe.getAddress` method:
```
Comment (germartinez) [inline pages/sdk/protocol-kit/guides/migrate-to-v5.md:87]:
```suggestion
You can predict the address of a Safe account before its deployment, as long as you are using Safe `v1.3.0` or greater, by replacing the `SafeFactory.predictSafeAddress` method with the `Safe.getAddress` method:
```

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md (hunk 3 above)
Comment (DaniSomoza) [reply pages/sdk/protocol-kit/guides/migrate-to-v5.md:69]: done!

File: pages/sdk/protocol-kit/guides/migrate-to-v5.md (hunk 4 above)
Comment (DaniSomoza) [reply pages/sdk/protocol-kit/guides/migrate-to-v5.md]: done!

**Review by dasanra — APPROVED** [review]

**Pull request merged**